pkg/autoscaler/scaling/autoscaler.go

```diff
@@ -183,6 +183,10 @@ func (a *autoscaler) Scale(logger *zap.SugaredLogger, now time.Time) ScaleResult
 	if spec.Reachable {
 		maxScaleDown = math.Floor(readyPodsCount / spec.MaxScaleDownRate)
 	}
+	// If MinNonZeroReplicas was set, we want to be able to scale to zero
+	if readyPodsCount == float64(a.deciderSpec.MinNonZeroReplicas) {
+		maxScaleDown = 0.
+	}
```

Comment:
If the Pods are reachable this could be pretty extreme especially if the annotation is set to a larger number of replicas.

Reply:
I was of a couple of minds here... on the one hand, if we say min=5, then we should skip directly from 5 -> 0 when scaling down. That's how I was initially approaching this (which also helps explain overriding initial scale). On the other, we could also interpret this as a "we'll scale up initially to 5, but then adjust the numbers down gradually if the load isn't there". I'm ok with either approach.